lib/util/timestamp.ts

```diff
@@ -0,0 +1,75 @@
+import { DateTime } from 'luxon';
+
+export type Timestamp = string & { __timestamp: never };
+
+const timezoneOffset = new Date().getTimezoneOffset() * 60000;
+
+const millenium = 946684800000; // 2000-01-01T00:00:00.000Z
+const tomorrowOffset = 86400000; // 24 * 60 * 60 * 1000;
+
+function isValid(date: DateTime): boolean {
+  const tomorrow = DateTime.now().toMillis() + tomorrowOffset; // 24 * 60 * 60 * 1000;
+  const ts = date.toMillis();
+  return date.isValid && ts > millenium && ts < tomorrow;
+}
+
+export function asTimestamp(input: unknown): Timestamp | null {
+  if (input instanceof Date) {
+    const date = DateTime.fromJSDate(input, { zone: 'UTC' });
+    if (isValid(date)) {
+      return date.toISO() as Timestamp;
+    }
+
+    return null;
+  }
+
+  if (typeof input === 'number') {
+    const millisDate = DateTime.fromMillis(input, { zone: 'UTC' });
+    if (isValid(millisDate)) {
+      return millisDate.toISO() as Timestamp;
+    }
+
+    const secondsDate = DateTime.fromSeconds(input, { zone: 'UTC' });
```

Comment:
do we need this?

Reply:
If some datasource is capable of providing timestamp in the millisecond format, I could imagine it's done in the seconds format as well